addons/survey/models/survey_user.py

```diff
@@ -41,18 +44,43 @@ class SurveyUserInput(models.Model):
     test_entry = fields.Boolean(readonly=True)
     # identification and access
     token = fields.Char('Identification token', default=lambda self: str(uuid.uuid4()), readonly=True, required=True, copy=False)
+    # no unique constraint, as it identifies a pool of attempts
+    attempts_pool_token = fields.Char('Attempts pool token', readonly=True, copy=False)
```

Comment:
I would name it invite_token, as it will be mainly granted through invite (either manual or automatic when finishing a course)... please don't :dagger: me .

Also to check: we should generate it only if necessary (aka, linked to a survey with limited number of attempts). Purpose is to avoid to bloat the database with unnecessary tokens. If people change their survey configuration after answers have been created, no need to change existing; only future answers will get the invite token.

Reply:
Should be good now (hopefully).